### PR TITLE
Add test files for CopulaABC

### PR DESCRIPTION
**Pull request by xuhuifan**
#### Summary:
Thanks, Henri. 
This is the first draft of Copula ABC method test file. These added three new files are for the method test. More complete version will come shortly. 
Xuhui

#### Please make sure
- You have updated the CHANGELOG.rst
- You have updated the documentation (if applicable)

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD3 (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: elfi/examples/CopulaABC/Inference_COPULA_ABC_inherit_rejection.py [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        gs[:, ii] = np.log(ghat(thetas[:, ii], margs[:, ii]))
+    gsum = np.sum(gs, axis=1)
+
+    return np.exp(temp+gsum-0.5*np.log(det(Lambdas)))
+
+
+
+def dimension_wise_dis(summaries, observed):
+    return abs(summaries - observed)
+
```
Comment (vuolleko) [inline elfi/examples/CopulaABC/Inference_COPULA_ABC_inherit_rejection.py]: You could define
```
def dimension_wise_dis(*simulated, observed):
    simulated = np.column_stack(simulated)
    observed = np.column_stack(observed)
    return abs(simulated - observed)
```
and then use `elfi.Discrepancy(dimension_wise_dis, ...)`

File: elfi/model/utils.py [hunk 2]
```diff
@@ -40,7 +40,10 @@ def distance_as_discrepancy(dist, *summaries, observed):
     # Ensure observed are 2d
     observed = np.concatenate([np.atleast_2d(o) for o in observed], axis=1)
     try:
-        d = dist(summaries, observed)
+        if dist.keywords['metric']=='dimension_wise':
+            d = abs(summaries-observed)
+        else:
+            d = dist(summaries, observed)
```
Comment (vuolleko) [inline elfi/model/utils.py]: Not required with the `dimension_wise_dis` above.

File: elfi/examples/CopulaABC/Inference_COPULA_ABC_inherit_rejection.py [hunk 3]
```diff
@@ -0,0 +1,259 @@
+import logging
```
Comment (vuolleko) [inline elfi/examples/CopulaABC/Inference_COPULA_ABC_inherit_rejection.py]: Inference methods should go under `elfi/methods/`

**Comment by xuhuifan** [comment]
Hi Henri, 

Sorry I have messed up about the push requests (there are several commits together). The current version is a bit messy. If so, can you reject this pull request and I can pull a new request. 

Xuhui

**Comment by vuolleko** [comment]
We tend to use the 'squash and merge' option for pull requests, so all your commits will be 'squashed' into a single merge commit.

File: elfi/methods/parameter_inference.py [hunk 4]
```diff
@@ -7,6 +7,9 @@
```
Comment (vuolleko) [inline elfi/methods/parameter_inference.py]: It's fine to keep the CopulaABC implementation in its own file. The old `parameter_inference.py` is getting a bit too long anyway (for my taste at least).

**Comment by vuolleko** [comment]
Please remove the `.DS_STORE` files from the repository.

**Comment by xuhuifan** [comment]
Hi Henri, 
It looks like the current version is a bit messy. How about I close this request and propose a new request, which is adding a CopulaABC method only?
Xuhui

**Pull request closed without merging**